superset/assets/src/visualizations/leaflet_map.js

```diff
@@ -372,6 +373,54 @@ function leafletmap(slice, payload) {
         renderBasicMap();
         renderPolygonLayer();
         addColumnsDropdownToMap();
+        addHeatMapColorLegend();
+    }
+
+    function addHeatMapColorLegend(){
+      var colname  = getSelectedColorColumn()
+      var col = colorCols[colname];
+      var minValue = col['operator'];
+      var maxvalue = col['sqlExpression'];
+      var legends = splitRangeIntoEqualParts(parseFloat(minValue), parseFloat(maxvalue), 10);
+
+      var legend = L.control({ position: 'topleft' });
+      legend.onAdd = () => {
+        var div
+        if($('#heat-map-color-legend').length > 0){
+          div = $('#heat-map-color-legend')[0];
+        } else {
+          div = L.DomUtil.create('div');
+          div.setAttribute('id', 'heat-map-color-legend')
+
+          //to show legend horizontally, un-comment below line
+          //div.classList.add("horizontal");
+
+        }
+        div.innerHTML = getHeatMapColorLegendHTML(colname, legends);
+        return div;
+      };
+      legend.addTo(mapInstance);
+    }
+
+    function getHeatMapColorLegendHTML(colname, legends){
+      var html = "<div class='percentage-value'>0%</div>"
+
+      for(var i=0; i<legends.length; i++){
+        html += "<div class='legend-div' id='legend_"+i+"' style='background-color:" + getColorForColumnVaule(colname, legends[i])+ "'></div>"
+      }
+      html += "<div class='percentage-value'>100%</div>"
+      return html;
+    }
+
+    function splitRangeIntoEqualParts(left, right, parts) {
+      var result = [],
+          delta = (right - left) / (parts - 1);
+      while (left < right) {
+          result.push(left);
+          left += delta;
+      }
+      result.push(right);
+      return result;
     }
 
```

Comment:
can we break this into a separate legend component and integrate it here

Reply:
created legend component separately and using